pkg/repository/repository.go

```diff
@@ -89,3 +89,18 @@ func (r Repository) IsEqual(otherRepo Repository) bool {
 func cleanPath(path string) string {
 	return strings.TrimSuffix(strings.TrimPrefix(path, "/"), ".git")
 }
+
+// GetAnonymizedRepo returns a clean repo url string without sensible information
+func (r Repository) GetAnonymizedRepo() string {
```

Comment:
would be possible to overwrite the `String()` method? or would it have other side effects? if it's possible, we would make sure the credentials are never printed

Reply:
you mean url.URL.String() in this package? the url prop for Repository is private so it will need a method to expose it, not sure i understand the benefits

Reply:
The current solution is error-prone in the sense that it's very easy to print `url.URL.String()` for a new developer and expose the credentials. If `Repository` has a `String` method that masks the credentials, no one will be able to print the credentials. Ideally, that means `URL` will be private

Reply:
`Repository` struct has already private url, so this is the only method exposing the url, and removing the user information if it has.

I will take a look and make some changes

Reply:
updated code with changes